config/webpack/partial/pwa.js

```diff
@@ -0,0 +1,151 @@
+"use strict";
+var path = require("path");
+var assign = require("lodash/assign");
+var archetype = require("../../archetype");
+var mergeWebpackConfig = archetype.devRequire("webpack-partial").default;
+var fileLoader = archetype.devRequire.resolve("file-loader");
+var webAppManifestLoader = require.resolve("web-app-manifest-loader");
+var SWPrecacheWebpackPlugin = require("sw-precache-webpack-plugin");
+var FaviconsWebpackPlugin = require("favicons-webpack-plugin");
+var AddManifestFieldsPlugin = require('../plugins/add-manifest-fields');
+
+var swConfigPath = path.resolve(process.cwd(), "config/sw-config.js");
+
+function getSWConfig() {
+  var swConfig;
+
+  try {
+    swConfig = require(swConfigPath);
+  } catch(err) {
+    swConfig = {};
+  }
+
+  return swConfig;
+}
+
+/**
+ * Takes a file path and returns a webpack-compatible
+ * filename descriptor matching the current naming schema
+ * @param  {string} filepath  original file path
+ * @return {string}           parsed file path
+ */
+function getHashedPath(filepath) {
+  var parsed = path.parse(filepath);
+  var name = parsed.name;
+  var ext = parsed.ext;
+  return name + '.[hash]' + ext;
+}
+
+/**
+ * Takes an array of strings representing scripts that
+ * a service worker will import and returns an object
+ * representing the webpack entry config. Each script
+ * will get its own entry point so they can be independantly
+ * imported by the service worker.
+ * @param  {array<string>} importScripts
+ * @param  {object}        current entry config
+ * @return {object}        new entry config
+ */
+function createEntryConfigFromScripts(importScripts, entry) {
+  // Handle the case where there might already be multiple
+  // entry points. If it is we create a new object with all
+  // existing entry points to avoid mutating the config. If its not,
+  // we assume its a string and use it as the bundle entry point.
+  var newEntry = typeof entry === "object"
+    ? Object.assign({}, entry)
+    : { bundle: entry };
+  return importScripts.reduce(function(acc, script) {
+    var name = path.parse(script).name;
+    acc[name] = script;
+    return acc;
+  }, newEntry);
+}
+
+module.exports = function () {
+  return function (config) {
+    var swConfig = getSWConfig();
+
+    if (!swConfig.manifest) {
+      return mergeWebpackConfig(config, {});
+    }
+
+    var manifestConfig = assign({
+      background: "#FFFFFF",
+      logo: "./images/electrode.png",
+      title: "Electrode",
+      short_name: "Electrode",
+      statsFilename: "../server/iconstats.json"
+    }, swConfig.manifest);
+
+    var cacheConfig = assign({
+      staticFileGlobs: [
+        "dist/js/*.{js,css}"
+      ],
+      stripPrefix: "dist/js/",
+      cacheId: "electrode",
```

Comment:
@ananavati they are config driven, we're using `assign` here with `swConfig.cache` so any values in the config will overwrite these defaults.